engine/engine/src/engine.cpp

```diff
@@ -1535,16 +1535,32 @@ namespace dmEngine
                     DM_PROFILE("Script");
 
                     // Script context updates
-                    if (engine->m_SharedScriptContext) {
+                    dmGameSystem::ScriptLibContext script_lib_context;
+                    script_lib_context.m_Factory  = engine->m_Factory;
+                    script_lib_context.m_Register = engine->m_Register;
+
+                    if (engine->m_SharedScriptContext)
+                    {
+                        script_lib_context.m_LuaState = dmScript::GetLuaState(engine->m_SharedScriptContext);
+                        dmGameSystem::UpdateScriptLibs(script_lib_context);
                         dmScript::Update(engine->m_SharedScriptContext);
-                    } else {
-                        if (engine->m_GOScriptContext) {
+                    }
+                     else
+                     {
+                        if (engine->m_GOScriptContext)
+                        {
+                            script_lib_context.m_LuaState = dmScript::GetLuaState(engine->m_GOScriptContext);
+                            dmGameSystem::UpdateScriptLibs(script_lib_context);
```

Comment:
We need to call this to keep track of all active requests. The load queue has a limited amount of pending requests, so we call the update function to dispatch initalized but not sent requests (preload queue does something similar)